### PR TITLE
cmake: update to cmake: 3.15.7

### PR DESCRIPTION
**Pull request by ioannisg**
Move cmake version to 3.15.7.
This is required for building Zephyr with TF-M support.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
cc @galak @carlescufi @nashif

File: shippable/install_source.sh
Comment (galak) [inline shippable/install_source.sh]: Probably best for this to be 3.15.6 ( I think that is the latest dot release )

Comment (carlescufi) [reply shippable/install_source.sh]: Agreed, we should use the latest stable in the series.

Comment (ioannisg) [reply shippable/install_source.sh]: Yeap, sure, I'll update this.

Comment (carlescufi) [reply shippable/install_source.sh]: Latest Ubuntu LTS comes with 3.16.3: https://packages.ubuntu.com/focal/cmake, so I think 3.15.6 is a good choice.

Comment (galak) [reply shippable/install_source.sh]: Looks like v3.15.7 based on the git tags for CMake repo.

Comment (nashif) [reply shippable/install_source.sh]: just make sure it works with our current tree and does not generate all kind of policy warnings that would force us to move minimum required version before we know it....

Comment (ioannisg) [reply shippable/install_source.sh]: Yeap, 3.15.7 exists in GH

File: shippable/install.sh
Comment (galak) [inline shippable/install.sh:149]: Not related to this PR, but do we know why imgtool isn't in https://raw.githubusercontent.com/JuulLabs-OSS/mcuboot/master/scripts/requirements.txt ?

**Review by galak — DISMISSED** [review]

**Comment by ioannisg** [comment]
Extra commit is for adding imgtool and lpc_checksum python packages, which is currently done in .buildkite as a temp-hack.

File: shippable/install.sh
Comment (ioannisg) [reply shippable/install.sh:149]: Good point, I was actually looking into the mcu boot requirements, and plan to check with David.

Comment (ioannisg) [reply shippable/install.sh:149]: @galak it is now: https://github.com/mcu-tools/mcuboot/blob/master/scripts/requirements.txt

Comment (ioannisg) [reply shippable/install.sh:149]: though the contents are the same :)

**Review by carlescufi — DISMISSED** [review]

File: shippable/install.sh
Comment (galak) [reply shippable/install.sh:149]: do you mind adding a commit to move to ` https://github.com/mcu-tools`

Comment (ioannisg) [reply shippable/install.sh:149]: yeap, sure, done

File: shippable/install.sh
Comment (galak) [inline shippable/install.sh]: We can remove this, it should be covered by the zephyr requirements.txt

**Comment by ioannisg** [comment]
Does this repo automatically dismiss reviews on pushing updates?

File: shippable/install.sh
Comment (ioannisg) [reply shippable/install.sh]: > We can remove this, it should be covered by the zephyr requirements.txt

Just to clarify @galak: why don't we do the same for imgtool?

One option would be to keep it here temporarily until it is passed in mcuboot requirements.txt.
But then, imgtool is used when we integrate with TFM (which uses it's own mcuboot, so then it "becomes" a requirement for zephyr)

**Comment by galak** [comment]
> Does this repo automatically dismiss reviews on pushing updates?

Not sure.  Maybe some new github feature.

**Comment by galak** [comment]
> Not sure. Maybe some new github feature.

looks like we have `Dismiss stale pull request approvals when new commits are pushed` set.

File: shippable/install.sh
Comment (ioannisg) [reply shippable/install.sh]: For now I kept imgtool, as it is not covered by neither mcuboot nor zephyr requirements.txt.

Comment (carlescufi) [reply shippable/install.sh]: I agree, why not add it to `requirements-extras.txt`? I mean, we are already installing it in CI via hack:
https://github.com/zephyrproject-rtos/zephyr/blob/master/.buildkite/run.sh#L58

Comment (carlescufi) [reply shippable/install.sh]: >Just to clarify @galak: why don't we do the same for imgtool?

`imgtool` is not in our requirements.txt today.

**Review by galak — APPROVED** [review]

File: shippable/install.sh
Comment (ioannisg) [reply shippable/install.sh]: > imgtool is not in our requirements.txt today.
@carlescufi i meant, why don't we add it in our requirements, as well?

Comment (galak) [reply shippable/install.sh]: @carlescufi my comment was w/regards to lpc_checksum being in requirements.txt (specifically requirements-extras.txt)

**Pull request merged**
